File: java/src/two_pointers/LC_167_TwoSumII.java
package two_pointers;
// https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/

public class LC_167_TwoSumII {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0, right = numbers.length - 1;
        while (left < right) {
            int sum = numbers[left] + numbers[right];
            if (sum == target)
                return new int[] { left + 1, right + 1 };
            if (sum > target)
                right--;
            else
                left++;
        }
        return new int[] { -1, -1 };
    }
}

File: python/two_pointers/LC_167_TwoSumII.py
# https://leetcode.com/problems/two-sum-ii-input-array-is-sorted/
from typing import List
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left, right = 0, len(numbers)-1
        while left < right:
            sum = numbers[left] + numbers[right]
            if (sum == target):
                return [left+1, right+1]
            if sum > target:
                right -=1
            else:
                left +=1
        return []
            